### PR TITLE
Add actions reducers

### PR DESCRIPTION
**Pull request by Piercel2022**
The following requirements are implemented in this pull request

- First of all - make sure that you keep state immutable in reducers!
- The default state in the books reducer changes from an empty array to an array with a few books(2)
- The Redux Provider and the store are imported in the main component of the  app.
- The React component responsible for adding new books, add functionality for "Add new" button click is added
- The React component responsible for removing books, add a functionality for "Remove" button click is added
- The React component responsible for categories, add a functionality for "Check status" button click is added

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ ReactDOM.render(
           path="/"
           exact
           element={(
+            // Import Redux Provider and the store in the main component of the app.
             <Provider store={store}>
               <Books />
```
Comment (tresorsawasawa) [inline src/index.js]: - [ ] As required for the project, kindly make sure the Main React component is wrapped in Provider and assigned `store`. I just noticed you didn't use the `App.js` as the main component, as is the case you should wrap all the code from `line 13 - 35`  in the `Provider`. All components are supposed to be handled with `Redux` and not only the `Books` component.

File: src/components/books/AddnewBook.js [hunk 2]
```diff
@@ -18,6 +17,7 @@ function AddnewBook() {
   const onClick = () => {
     setHide(!hide);
   };
+  const addBook = () => { };
   const onSubmit = (e) => {
     const title = bookTitle.current.value;
     const enteredCategorie = categorie;
```
Comment (tresorsawasawa) [inline src/components/books/AddnewBook.js]: - [ ] Kindly make sure your form to `Add` a new book is available and **ADDING** and **REMOVING** new books are possible as required for the project.

File: src/components/categories/CategorieList.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 function CategoriesList(props) {
   const { hide } = props;
   const { selectHandler } = props;
   return (
     <ul className={hide ? classes.ul : classes.hidden}>
-      <button type="button" name="Action" onClick={selectHandler}>Check Status</button>
+      <button type="button" name="Check Status" onClick={selectHandler}>Check Status</button>
     </ul>
   );
 }
```
Comment (tresorsawasawa) [inline src/components/categories/CategorieList.js]: - [ ] Kindly make sure the "Check status" button is available on the categories page and it is working as expected i.e it should dispatch the `underConstruction` action creator.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Piercel2022 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mateo951 — APPROVED** [review]
### Hi @Piercel2022, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🔥 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**